packages/client/hmi-client/src/components/projects/Project.vue

```diff
@@ -0,0 +1,168 @@
+<script setup lang="ts">
+import API from '@/api/api';
+import { ref } from 'vue';
+
+const props = defineProps({
+	id: {
+		type: String,
+		required: true
+	}
+});
+
+const name = ref('');
+const description = ref('');
+const timestamp = ref('');
+
+// refactor as a composable?
+async function getProject() {
+	return API.get(`/projects/${props.id}`);
+}
+
+getProject().then((response) => {
+	const data = response.data;
+	name.value = data.name;
+	description.value = data.description;
+	timestamp.value = data.timestamp;
+});
```

Comment:
Slight preference if we marshal the response into a Project type object rather than 3-independent ref-variables, since you have created a Project-type anyway.

Reply:
Yes, I wanted to do that as well but was just running into some trouble (mainly just unfamiliarity with typescript). I think it would be a good idea to move all the http functions into a composable, so you think this would be a good candidate for something to work on in a future pass?

Reply:
Just to reduce the queue a bit I pushed in the typing changes, you'd do something like this - mainly through the declaration of the return type `Promise<XYZ>`

```
const project = ref<Project>({
	id: '',
	name: '',
	description: '',
	timestamp: ''
});

// refactor as a composable?
async function getProject(): Promise<Project> {
	const response = await API.get(`/projects/${props.projectId}`);
	return response.data;
}

onMounted(async () => {
	project.value = await getProject();
});
```